go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/tektoncd/pipeline
 
-go 1.19
+go 1.21
+
+toolchain go1.21.0
```

Comment:
Did this come from running `./hack/update-codegen.sh`? If not, may be we can rever this and let dependabots bump it for us.